### PR TITLE
feat(webapp): option to select all election years and last election

### LINKED ISSUE #125: Option to select all election years and last election
image

### PR DESCRIPTION
**Pull request by warnerHurtado**
### What does this PR do?

- Resolve #125 

### Steps to test

1. run the project and open localhost:3000 in your browser 
2. Click on income on the sidebar
3. Click on one Election radio button
4. Now you can see all by default in year select and election select you can see all election

#### CheckList

- [X] Follow proper Markdown format
- [X] The content is adequate
- [X] The content is available in both english and spanish
- [X] I Ran a spell check

image

### CODE REVIEW

File: webapp/src/hooks/customHooks/useIncomeReportState.js [hunk 1]
```diff
@@ -281,8 +287,7 @@ const useIncomeReportState = () => {
 
       setElectionsByYearList(electionsByYearData.eden_historic_election)
     }
-
-    if (!electionsByYearData?.eden_historic_election[0]) {
+    if (electionsByYearList[0]?.election !== electionRoundSelect) {
```
Comment (leisterfrancisco) [inline webapp/src/hooks/customHooks/useIncomeReportState.js:290]: blank line before if statement

**Review by leisterfrancisco — CHANGES_REQUESTED** [review]
Take a look at my comment

File: webapp/src/hooks/customHooks/useIncomeReportState.js [hunk 2]
```diff
@@ -216,10 +216,16 @@ const useIncomeReportState = () => {
   const [loadElectionsByYear, { data: electionsByYearData }] = useLazyQuery(
     GET_ELECTIONS_BY_YEAR,
     {
-      variables: {
-        minDate: `${electionYearSelect}-01-01`,
-        maxDate: `${electionYearSelect}-12-31`
-      }
+      variables:
+        electionYearSelect === 'All'
```
Comment (xavier506) [inline webapp/src/hooks/customHooks/useIncomeReportState.js:220]: is this text being translated on the UI?

**Comment by xavier506** [comment]
@warnerHurtado 

Please resolve conflicts

**Review by xavier506 — APPROVED** [review]
LGTM

**Pull request merged**